flink-connectors/pom.xml

```diff
@@ -85,6 +86,17 @@ under the License.
 				<module>flink-connector-kinesis</module>
 			</modules>
 		</profile>
+
+		<!--
+			Since Elasticsearch 5.x requires Java 8 at a minimum, we use this profile
+			to include it as part of Java 8 builds only.
+		-->
+		<profile>
+			<id>include-elasticsearch5</id>
```

Comment:
Can you make this a profile that activates itself automatically when java 8 is available?
http://maven.apache.org/guides/introduction/introduction-to-profiles.html